scripts/maintenance/NULL-assign

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/awk -f
+#
+## Copyright (C) 1996-2020 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+/NULL/ {
+  # NULL is deprecated in C++11 by nullptr.
+  if ( FILENAME ~ /(\.cci?|src\/|tools\/)$/ ) {
+    gsub(/[ ]=[ ]?NULL[;]/, " = nullptr;");
+    gsub(/[ ]=[ ]?NULL[,]/, " = nullptr,");
+    gsub(/[ ]=[ ]?NULL[)]/, " = nullptr)");
```

Comment:
Why three conditions and not a capture group or a word delimiter?

Reply:
Simplicity. This is closest to declarative replacement that I know of. We can add and remove individual replacements as special cases are found in future.

Reply:
Also, it seems only gawk supports back references to capture groups. Without that we loose the patterns terminating delimiter.